docs/provision_hypershift_clusters_by_mce.md

```diff
@@ -129,31 +129,202 @@ Upon scaling up a NodePool, a Machine will be created, and the CAPI provider wil
 ```bash
-$ oc apply -f - <<EOF
-apiVersion: cluster.open-cluster-management.io/v1alpha1
-kind: HypershiftDeployment
+$ oc patch multiclusterengine <mce_name> --type=merge -p '{"spec":{"overrides":{"components":[{"name":"assisted-service","enabled": true}]}}}'
+```
+
+3. Create the agentserviceconfig object. Double check the `ISO_URL` at https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/${OCP_VERSION}/latest.
+```bash
+export DB_VOLUME_SIZE="10Gi"
+export FS_VOLUME_SIZE="10Gi"
+export OCP_VERSION="4.10"
+export ARCH="x86_64"
+export OCP_RELEASE_VERSION=$(curl -s https://mirror.openshift.com/pub/openshift-v4/${ARCH}/clients/ocp/latest-${OCP_VERSION}/release.txt | awk '/machine-os / { print $2 }')
+export ISO_URL="https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/${OCP_VERSION}/latest/rhcos-${OCP_VERSION}.3-${ARCH}-live.${ARCH}.iso"
+export ROOT_FS_URL="https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/${OCP_VERSION}/latest/rhcos-live-rootfs.${ARCH}.img"
+
+envsubst <<"EOF" | oc apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: AgentServiceConfig
 metadata:
-  name: hypershift-demo
+ name: agent
+spec:
+  databaseStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: ${DB_VOLUME_SIZE}
+  filesystemStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: ${FS_VOLUME_SIZE}
+  osImages:
+    - openshiftVersion: "${OCP_VERSION}"
+      version: "${OCP_RELEASE_VERSION}"
+      url: "${ISO_URL}"
+      rootFSUrl: "${ROOT_FS_URL}"
+      cpuArchitecture: "${ARCH}"
+EOF
+```
+
+4. Wait for the assisted-service pod to be ready.
+```bash
+until oc wait -n multicluster-engine $(oc get pods -n multicluster-engine -l app=assisted-service -o name) --for condition=Ready --timeout 10s >/dev/null 2>&1 ; do sleep 1 ; done
+```
+
+###### Create bare metal host and agent to be used as a worker node on hosting cluster
+
+The number of `BareMetalHost` resources should match the `agent` namespace should match the number of replica in `NodePool`. Follow https://github.com/openshift/hypershift/blob/main/docs/content/how-to/agent/create-agent-cluster.md#adding-a-bare-metal-worker for creating `BareMetalHost` and `agent` resources. Stop when `agent` resources are created. Skip updating the nodepool part of the documentation. Note the namespce for the `agent` resources. This namespace will be used as `agentNamespace` in `HostedCluster` resource in the next section.
+
+
+###### Provision a hosted cluster
+
+Create `HostedCluster` and `NodePool` on the MCE cluster. These will be referenced by `HypershiftDeployment` to provision the hosted cluster on the target hosting cluster. We are going to create the `HostedCluster`, `NodePool` and  `HypershiftDeployment` all in `default` namespace on the MCE cluster. On the hosting cluster, hypershift deployment will create `HostedCluster` and `NodePool` in `clusters` namespace.
```

Comment:
Were you able to make this work? My understanding is it will only work for the hub.